src/python/tests/core/bot/tasks/train_gradientfuzz_task_test.py

```diff
@@ -0,0 +1,249 @@
+    self.dataset_dir = os.path.join(
+        self.home_dir, constants.DATASET_DIR,
+        self.fuzzer_name + run_constants.CORPUS_SUFFIX)
+    self.run_dir = os.path.join(
+        self.home_dir, constants.MODEL_DIR,
+        constants.NEUZZ_ONE_HIDDEN_LAYER_MODEL,
+        self.fuzzer_name + run_constants.RUN_NAME_SUFFIX)
+    self.temp_dir = tempfile.mkdtemp()
+    self.binary_path = os.path.join(GRADIENTFUZZ_TESTING_DIR, TESTING_BINARY)
+
+    # TODO(ryancao): Fix env var name!
+    os.environ['TEST_BINARY_PATH'] = self.binary_path
+    os.environ['FUZZ_INPUTS_DISK'] = self.temp_dir
+
+    test_helpers.patch(self, [
+        'bot.tasks.train_gradientfuzz_task.get_corpus',
+        'bot.tasks.train_gradientfuzz_task.upload_model_to_gcs',
+        'build_management.build_manager.setup_build'
+    ])
+
+    self.mock.upload_model_to_gcs.return_value = True
+    self.mock.get_corpus.side_effect = self.mock_get_corpus
+    self.mock.setup_build.side_effect = self.mock_build_manager
+
+  def mock_build_manager(self):
+    """
+    Just sets the 'APP_PATH' environment variable.
+    """
+    os.environ['APP_PATH'] = self.binary_path
+
+  def mock_get_corpus(self, corpus_directory, _):
+    """
+    Copy over training corpus to temp dir.
+    """
+    train_files = glob.glob(os.path.join(self.corpus_dir, '*'))
+    for train_file in train_files:
+      target_path = os.path.join(corpus_directory, os.path.basename(train_file))
+      self.assertTrue(shell.copy_file(train_file, target_path))
+    return True
+
+  def tearDown(self):
+    shell.remove_directory(os.path.join(self.home_dir, constants.DATASET_DIR))
+    shell.remove_directory(os.path.join(self.home_dir, constants.MODEL_DIR))
+    shell.remove_directory(os.path.join(self.home_dir, constants.GENERATED_DIR))
+    shell.remove_directory(self.temp_dir)
+
+  def test_train_gradientfuzz(self):
+    """
+    Generate input/output pairs, then train GradientFuzz
+    model on a simple corpus.
+    """
+    train_gradientfuzz_task.execute_task(self.fuzzer_name, self.job_type)
+
+    # Asserts that directories were created.
+    inputs = os.path.join(self.dataset_dir, constants.STANDARD_INPUT_DIR)
+    labels = os.path.join(self.dataset_dir, constants.STANDARD_LABEL_DIR)
+    self.assertTrue(os.path.isdir(inputs))
+    self.assertTrue(os.path.isdir(labels))
+    self.assertTrue(os.path.isdir(self.run_dir))
+
+    # Attempts to load in latest model.
+    ckpt_path = tf.train.latest_checkpoint(self.run_dir)
+    model_config_path = os.path.join(self.run_dir, constants.CONFIG_FILENAME)
+    model_config = json.load(open(model_config_path, 'r'))
+    model = models.make_model_from_layer(
+        constants.ARCHITECTURE_MAP[model_config['architecture']],
+        model_config['output_dim'],
+        model_config['input_shape'],
+        hidden_layer_dim=model_config['num_hidden'])
+    model.load_weights(ckpt_path).expect_partial()
+
+    # TODO(ryancao): Check model accuracy!
```

Comment:
I'll do another quick pass once the remaining tests are added. Do you still have plans to check the accuracy? I'm not sure that's necessary as an integration test. If there's anything here you don't end up adding, I think it's probably best to remove the TODOs. The test with a corpus you were planning makes sense.

Reply:
Yup -- removed the TODOs!